### ISSUE TITLE
No image appears with nbagg

### ISSUE THREAD

**Issue by beojan** [matplotlib/matplotlib#3962]
See issue https://github.com/ipython/ipython/issues/7351

**Comment by WeatherGod** [comment]
Please post an SSCCE: http://sscce.org/

Also, I assume you mean "appears" instead of "applies"?

**Comment by beojan** [comment]
![ipython nbagg](https://cloud.githubusercontent.com/assets/3727925/5597571/89a823fe-929c-11e4-851c-614786aa9974.png)


```python
import matplotlib
import numpy as np
matplotlib.use('nbagg')
import matplotlib.pyplot as plt
x = np.linspace(0,10,10000)
y = np.sin(x)
plt.plot(x,y)
plt.show()
```

**Comment by WeatherGod** [comment]
Also, which version of matplotlib and ipython are you using, and on which
system (linux? mac? windows?). Also, it might be relevant to know which
version of tornado you are using

On Fri, Jan 2, 2015 at 11:30 AM, Beojan Stanislaus <notifications@github.com

**Comment by beojan** [comment]
Python 3.4 on Debian with matplotlib and ipython from git. with tornado 4.0.2

**Comment by tacaswell** [comment]
This looks like a duplicate of https://github.com/matplotlib/matplotlib/issues/3606

**Comment by tacaswell** [comment]
I can not replicate this with ipython (2.3.1) + mpl (1.4.2) installed from anaconda, trying with master now.

**Comment by tacaswell** [comment]
- mpl master + ipython 2.3.1 work
 - mpl master + ipython master does not

So something changed over on their end.

**Comment by tacaswell** [comment]
and mpl 1.4.2 + ipython master does not work

**Comment by beojan** [comment]
http://ipython.org/ipython-doc/dev/development/messaging.html
The IPython messaging api has changed from version 4.1 (in IPython 2.3) to 5 (in IPython 3.0 [master])

**Comment by tacaswell** [comment]
We should be insulated from that an we are using `Comm` objects to deal with the actual wire transfer.  I suspect something has become crossed in how the comm is built/registered.

This fits why there is no image (the send on the mpl side is talking to a non-existent comm) and the error messages from the python side on mouse events (it is also talking to a non-existent comm).

**Comment by tacaswell** [comment]
closed by #4002
